src/main/java/com/lpnu/shaggybeavers/service/impl/UserServiceImpl.java

```diff
@@ -17,4 +18,10 @@ public class UserServiceImpl extends CRUDServiceImpl<User,Long> implements UserS
     protected JpaRepository<User, Long> getRepository () {
         return this.repository;
     }
+
+    @Override
+    public User findByEmail(String email) {
+        return repository.findByEmail(email)
+                .orElseThrow(NotExistsUserException::new);
```

Comment:
It is un flexible creating exception under each service, you can reuse NotExistObjectException with specif message